### PR TITLE
Serde all the things

### PR DESCRIPTION
**Pull request by AhmedSoliman**
Serde all the things

### CODE REVIEW

File: crates/types/src/identifiers.rs [hunk 1]
```diff
@@ -44,7 +44,7 @@ pub type PeerId = u64;
 )]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 #[display(fmt = "e{}", _0)]
-pub struct LeaderEpoch(u64);
+pub struct LeaderEpoch(u32);
```
Comment (slinkydeveloper) [inline crates/types/src/identifiers.rs:47]: This one seems unrelated to this PR?

**Review by slinkydeveloper — APPROVED** [review]
LGTM

File: crates/types/src/identifiers.rs (hunk 1 above)
Comment (AhmedSoliman) [reply crates/types/src/identifiers.rs]: Correct, it was pulled down by mistake.
